CalibTracker/SiStripHitEfficiency/plugins/HitEff.cc

```diff
@@ -592,7 +613,56 @@ void HitEff::analyze(const edm::Event& e, const edm::EventSetup& es) {
             }
           }
 
-          if (!tmpTmeas.empty()) {
+          //Test for two consecutive missing hits
+          if (int(TKlayers) - int(prev_TKlayers) == -3) {
+            foundConsMissingHits = true;
+            const DetLayer* detlayer = itm->layer();
+            const LayerMeasurements layerMeasurements{*measurementTrackerHandle, *measurementTrackerEvent};
+            const TrajectoryStateOnSurface tsos = itm->updatedState();
+            std::vector<DetLayer::DetWithState> compatDets = detlayer->compatibleDets(tsos, *thePropagator, *estimator);
+
+            if (misLayer > k_LayersStart && misLayer < k_LayersAtTOBEnd && previousMisLayer > k_LayersStart &&
+                previousMisLayer < k_LayersAtTOBEnd) {  //Barrel case
+              std::vector<BarrelDetLayer const*> barrelTIBLayers =
+                  measurementTrackerHandle->geometricSearchTracker()->tibLayers();
+              std::vector<BarrelDetLayer const*> barrelTOBLayers =
+                  measurementTrackerHandle->geometricSearchTracker()->tobLayers();
+              if (misLayer > k_LayersStart && misLayer < k_LayersAtTIBEnd) {
+                const DetLayer* tibLayer = barrelTIBLayers[misLayer - k_LayersStart - 1];
+                const DetLayer* prevTibLayer = barrelTIBLayers[previousMisLayer - k_LayersStart - 1];
+
+                tmpTmeas = layerMeasurements.measurements(*tibLayer, tsos, *thePropagator, *estimator);
+                prev_tmpTmeas = layerMeasurements.measurements(*prevTibLayer, tsos, *thePropagator, *estimator);
+              } else if (misLayer > k_LayersAtTIBEnd && misLayer < k_LayersAtTOBEnd) {
```

Comment:
Also?
```suggestion
              } else if (misLayer > k_LayersAtTIBEnd && misLayer <= k_LayersAtTOBEnd) {
```
(and the same here below)

Reply:
The reason for it to be strictly lower than k_LayersAtTOBEnd is because of the size of the array :

const DetLayer* prevTobLayer = barrelTOBLayers[previousMisLayer - k_LayersAtTIBEnd - 1];

Since the previous missed layer is numbered as the missed layer + 1, this would cause to point outside the array.